docusaurus/docs/iOS/guides/working-with-messagelist-layout.md

```diff
@@ -0,0 +1,128 @@
+---
+title: Working with MessageList Layout
+---
+
+## `ChatMessageLayoutOptionsResolver`
+
+Almost anything related to the layout and appearance of the message cell can be done by subclassing [`ChatMessageLayoutOptionsResolver`](../ReferenceDocs/Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageLayoutOptionsResolver.md). 
+On  `ChatMessageLayoutOptionsResolver` there is intance property `options`([ChatMessageLayoutOptions](../ReferenceDocs/Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageLayoutOptions.md)) which holds available options for the cell. 
+See guides below for concrete examples.
+
+## Left-aligning all messages
+ To left-align all messages inside MessageList, you first need to create your custom left-aligned `MessageLayoutOptionsResolver` like this:
+ 
+ ```swift
+ class LeftAlignedMessageLayoutOptionsResolver: ChatMessageLayoutOptionsResolver {
+     override func optionsForMessage(at indexPath: IndexPath, in channel: _ChatChannel<NoExtraData>, with messages: AnyRandomAccessCollection<_ChatMessage<NoExtraData>>) -> ChatMessageLayoutOptions {
+         // Get options for the message at given indexpath to change it. 
+         var options = super.optionsForMessage(at: indexPath, in: channel, with: messages)
+         // First it's needed to disable the flipping of sides when messages is sent from current user
+         options.remove(.flipped)
+         // After that we need to ensure that for current user there will be avatar included in the message.
+         options.insert(.avatar)
+         // If you want, you can include the author name for the message as well.
+         options.insert(.authorName)
+         return options
+     }
+ }
+ ```
+ 
+ When your custom `ChatMessageLayoutOptionsResolver` is created, all you need to do is to set this class to `Components` before displaying the `MessageList`:
+ 
+ ```swift
+ Components.default.messageLayoutOptionsResolver = LeftAlignedMessageLayoutOptionsResolver()
+ ```
+ Here is the result: 
```

Comment:
would be better if we did a default/custom comparison, side by side, like we do in UI customization guide

Reply:
At some point, we should make a react component that we can add through .mdx. It will make it easier to add and maintain.